### PR TITLE
 👌 Removing deepcopy

### PR DESCRIPTION
**Pull request by danwos**
Tasks to do:
* [x] Remove deepcopy calls for filtering functions
* [ ] ~~Find alternative to secure need-data from manipulations~~ (Done by #1018)

Looks like this little change brings up to 80% better performance in a build.
Huge project got down to 5min from 28min.
Also memory consumption is cut down to 50-60%.

### CODE REVIEW

File: sphinx_needs/directives/needpie.py [hunk 1]
```diff
@@ -156,7 +155,8 @@ def process_needpie(app: Sphinx, doctree: nodes.document, fromdocname: str, foun
                 # execute filter_func code
                 # Provides only a copy of needs to avoid data manipulations.
                 context = {
-                    "needs": copy.deepcopy(need_list),
+                    # "needs": copy.deepcopy(need_list),
```
Comment (chrisjsewell) [inline sphinx_needs/directives/needpie.py:158]: you can remove the commented out code, we have the git history for that 😄 
```suggestion
```

File: sphinx_needs/filter_common.py [hunk 2]
```diff
@@ -167,7 +166,8 @@ def process_filters(
     else:
         # Provides only a copy of needs to avoid data manipulations.
         context = {
-            "needs": copy.deepcopy(all_needs_incl_parts),
+            # "needs": copy.deepcopy(all_needs_incl_parts),
```
Comment (chrisjsewell) [inline sphinx_needs/filter_common.py:169]:
```suggestion
```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

**Comment by danwos** [comment]
Right, also forget about the old code comments

**Review by chrisjsewell — APPROVED** [review]
Cheers

**Pull request merged**